File: ios/Model/TLToDo.swift
//
//  TLToDo.swift
//  Runner
//
//  Created by 林 明虎 on 2024/11/23.
//


struct TLToDo: Identifiable, Codable {
    var id: String
    var title: String
    var isChecked: Bool
    var steps: [TLStep]
}

File: lib/model/todo/tl_todo.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import './tl_step.dart';

part '../generate/tl_todo.freezed.dart';
part '../generate/tl_todo.g.dart';

// $ dart run build_runner build

@freezed
class TLToDo with _$TLToDo {
  const factory TLToDo({
    required String id,
    required String workspaceID,
    required bool isInToday,
    @Default(false) bool isChecked,
    required String content,
    @Default([]) List<TLStep> steps,
  }) = _TLToDo;

  factory TLToDo.fromJson(Map<String, dynamic> json) => _$TLToDoFromJson(json);
}
